src/DataSourceProvider.vue

```diff
@@ -1,19 +1,368 @@
+
+      return allDataSources;
+    },
+    getOtherAppsDataSources: function(dataSources) {
+      return dataSources.filter(dataSource => {
+        return this.currentAppDataSources.findIndex(currDS => currDS.id === dataSource.id) === -1;
+      });
+    },
+    formatDataSourceOption: function(data) {
+      const { id, name, text } = data;
+
+      return `${name || text} ID: ${id}`;
+    },
+    customDataSourceSearch: function(condition, data) {
+      // Return of this function should be the same as the array.filter function
+      // If there are no search terms, return all of the data
+      if (!condition) {
+        return true;
+      }
+
+      const term = condition.toUpperCase();
+      const name = data.name.toUpperCase();
+      const id = data.id.toString();
+
+      // Search both by name and id
+      if (name.indexOf(term) > -1 || id.indexOf(term) > -1) {
+        return true;
+      }
+
+      // Return `false` if the term should not be displayed
+      return false;
+    },
+    setDataSource: function(dataSource) {
+      if (dataSource) {
+        this.selectedDataSource = dataSource;
+
+        Fliplet.Widget.emit('showColumns', {
+          columns: dataSource.columns,
+          id: dataSource.id
+        });
+
+        this.$emit('onDataSourceSelect', dataSource);
+      }
+    },
+    sortDataSourceEntries: function(dataSources) {
+      const copyDataSources = [...dataSources];
+
+      if (copyDataSources[0].options) {
+        copyDataSources[0].options.sort(this.sortArray);
+        copyDataSources[1].options.sort(this.sortArray);
+      } else {
+        copyDataSources.sort(this.sortArray);
+      }
+
+      return copyDataSources;
+    },
+    sortArray: function(a, b) {
+      // Sort data source array by name
+      // Send names that starts with number to the end of the list
+      const startsWithAlphabet = /^[A-Z,a-z]/;
+      let aValue = a.name ? a.name.toUpperCase() : '}';
+      let bValue = b.name ? b.name.toUpperCase() : '}';
+
+      if (!startsWithAlphabet.test(bValue)) {
+        bValue = `{${bValue}`;
+      }
+
+      if (!startsWithAlphabet.test(aValue)) {
+        aValue = `{${aValue}`;
+      }
+
+      if (aValue < bValue) {
+        return -1;
+      }
+      if (aValue > bValue) {
+        return 1;
+      }
+
+      return 0;
+    },
+    viewDataSource: function() {
+      Fliplet.Studio.emit('overlay', {
+        name: 'widget',
+        options: {
+          size: 'large',
+          package: 'com.fliplet.data-sources',
+          title: 'Edit Data Sources',
+          classes: 'data-source-overlay',
+          data: {
+            context: 'overlay',
+            dataSourceId: this.selectedDataSource.id
+          }
+        }
+      });
+    }
   },
   components: {
-    FirstComponent
+    Select2
+  },
+  mounted: function() {
+    this.initProvider();
+
+    const $vm = this;
+
+    // Transfer selected DataSource id to the parent
+    Fliplet.Widget.onSaveRequest(function() {
```

Comment:
Arrow functions can be used to run a function without changing its `this` context.

```suggestion
    Fliplet.Widget.onSaveRequest(() => {
```